File: nettest-shared/src/main/java/at/alladin/nettest/shared/berec/berec/collector/api/v1/dto/measurement/detail/DetailMeasurementGroupItem.java
package at.alladin.nettest.shared.berec.collector.api.v1.dto.measurement.detail;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * This class specifies a single detail item with key, translated title, value and the unit of the value.
 * 
 * @author alladin-IT GmbH (bp@alladin.at)
 *
 */
@io.swagger.annotations.ApiModel(description = "This class specifies a single detail item with key, translated title, value and the unit of the value.")
@JsonClassDescription("This class specifies a single detail item with key, translated title, value and the unit of the value.")
public class DetailMeasurementGroupItem {
	
	/**
	 * The key is the path to the chosen field inside the data model, e.g. "client_info.app_version_name".
	 */
	@io.swagger.annotations.ApiModelProperty("The key is the path to the chosen field inside the data model, e.g. \"client_info.app_version_name\".")
	@JsonPropertyDescription("The key is the path to the chosen field inside the data model, e.g. \"client_info.app_version_name\".")
	@Expose
	@SerializedName("key")
	@JsonProperty("key")
	private String key;
	
	/**
	 * The translated title of this item.
	 */
	@io.swagger.annotations.ApiModelProperty("The translated title of this item.")
	@JsonPropertyDescription("The translated title of this item.")
	@Expose
	@SerializedName("title")
	@JsonProperty("title")
	private String title;
	
	/**
	 * The actual value of this item in the given unit.
	 */
	@io.swagger.annotations.ApiModelProperty("The actual value of this item in the given unit.")
	@JsonPropertyDescription("The actual value of this item in the given unit.")
	@Expose
	@SerializedName("value")
	@JsonProperty("value")
	private String value;
	
	/**
	 * The unit of the value.
	 */
	@io.swagger.annotations.ApiModelProperty("The unit of the value.")
	@JsonPropertyDescription("The unit of the value.")
	@Expose
	@SerializedName("unit")
	@JsonProperty("unit")
	private String unit;
	
}

File: measurement-agent-swift/MeasurementAgentKit/Sources/DTO/Result/Detail/DetailMeasurementGroupItem.swift
// MeasurementAgentKit: DetailMeasurementGroupItem.swift, created on 24.07.19
/*******************************************************************************
 * Copyright 2019 Benjamin Pucher (alladin-IT GmbH)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ******************************************************************************/

import Foundation

/// This class specifies a single detail item with key, translated title, value and the unit of the value.
public class DetailMeasurementGroupItem: Codable {

    /// The key is the path to the chosen field inside the data model, e.g. "device_info.model".
    public var key: String?

    /// The translated title of this item.
    public var title: String?

    /// The actual value of this item in the given unit.
    public var value: String?

    /// The unit of the value.
    public var unit: String?

    ///
    enum CodingKeys: String, CodingKey {
        case key
        case title
        case value
        case unit
    }
}
